### PR TITLE
Add "Continue" method for config file patching

### LINKED ISSUE #1122: Add multi-file configuration support
Pelican Service:
- [X] Client
- [ ] Plugin
- [X] Registry
- [X] Director
- [X] Origin
- [X] Cache
- [ ] Other (please give the detail)

**Is your feature request related to a problem? Please describe.**
In Kustomize-based Kubernetes deployments, we make heavy use of environment variables because it's easier to layer configuration between various bases and specific instances (e.g., all OSDF origins should have the same federation discovery URL). However, env var config is harder to read and makes it harder to implement some of the more interesting features like multiple exports.

**Describe the solution you'd like**
Ideally this would come in the form of files in `config.d` directories (yes, more than one!) that are parsed in lexicographic order. We make use of multiple `config.d` directories as packagers: we could provide a default federation discovery URL in `/usr/share/pelican/config.d` for the OSDF in the OSDF packaging, leaving `/etc/pelican/config.d` as the location for administrators to add their own config.

### PR DESCRIPTION
**Pull request by jhiemstrawisc**
This adds the ability to define "Continue" directories in your config, where doing so causes Pelican to merge in any additional files in that directory. For example, if my `/etc/pelican/pelican.yaml` has the line `Continue: /foo`, this addition causes Pelican to merge any extra yamls located in `/foo`. If `/foo/config.yaml` further defines `Continue: /bar`, Pelican will then look to `/bar` for extra config.

Config files in a "Continue" directory are parsed according to lexicographical order. This does not currently support adding entries to viper slices, so, for example, if `/etc/pelican/pelican.yaml` defines:
```
Origin:
  Exports:
    - < values for export 1 >
Continue: /foo
```

and `/foo/config.yaml` defines:
```
Origin:
  Exports:
    - < values for export 2 >
```

the exports slice will be overwritten with export 2, instead of appended to.

Closes #1122

### CODE REVIEW

**Comment by bbockelm** [comment]
I would like to see some discussion of the config name - I don't find "Continue" to be all that self-descriptive versus, say, "ConfigDir".

What are the semantics around recursion?  Can you have a config directory parameter discovered inside a config directory?

Should this be single-valued or a list?

**Comment by jhiemstrawisc** [comment]
I went with "Continue" because that's the same key used by XRootD, and it seemed like @brianhlin was asking for the same behavior. For people who are already familiar with XRootD, this should feel natural.

For your second question, if a `Continue` directory includes a config file that has another `Continue`, this will follow that chain until we enter a directory with config files that don't define a `Continue`.

For your last question, making this a list makes things a bit more nuanced and less intuitive. In that case, we'd have to decide whether to perform a depth-first or breadth-first crawl -- if I configure:
```
Continue: ["/foo", "/bar"]
```
and both of these directories contain further continues, do I first parse everything in `/foo` and then in all of `/foo`'s continue directories? Or do I do `/foo` and then `/bar` and then move back to `/foo`'s specified continue directories, etc etc. I favor the simplicity of the approach as it stands.

**Comment by jhiemstrawisc** [comment]
If we don't like `Continue`, I'll propose `ContinuedConfigLocation`. Everywhere else in our config, we use "Location" instead of "Dir", and I don't think `ConfigLocation` by itself captures that this is a type of "goto" statement.

**Comment by brianhlin** [comment]
> I went with "Continue" because that's the same key used by XRootD, and it seemed like @brianhlin was asking for the same behavior. For people who are already familiar with XRootD, this should feel natural.

As discussed during standup, let's not base our model on XRootD's config engine. I'd strongly prefer a `ConfigLocations` approach rather than thinking with continues but that does lead to interesting questions like Brian's about recursion and general Pelican config behavior (can we append to lists?).

File: config/config.go
Comment (joereuss12) [inline config/config.go:969]: Update this comment to say "ContinuedConfigLocation" key or whatever the term is decided to be.

File: config/config_test.go
Comment (joereuss12) [inline config/config_test.go:198]: Can you add a comment header here describing the test? Just want to stay consistent with everything

File: config/config_test.go
Comment (joereuss12) [inline config/config_test.go:304]: What if a user tries to continue a config dir that is empty or contains "config" files that are not `yaml`'s? I think we might want test cases for that.

**Review by joereuss12 — CHANGES_REQUESTED** [review]
Just have some small items but otherwise it LGTM

**Comment by jhiemstrawisc** [comment]
After feedback, the current mechanism works with the `ConfigLocations` config key and sets a list of extra dirs. For example:
```
ConfigLocations: ["/dir1", "/dir2"]
```
First we load all the configs according to lexicographical order in `/dir1` and then `/dir2`. There is no recursion into subdirectories. Config keys that are redefined are patched over, with the last config loaded winning.

File: docs/parameters.yaml
Comment (haoming29) [inline docs/parameters.yaml:39]: Are we still handling recursion here? Would it worth mentioned the behavior of nested `ConfigLocations`?

File: config/config.go
Comment (haoming29) [inline config/config.go:855]: For better readability, can we pull the body of this if statement out by simply returning the function if the length of the slice is 0?

File: config/config.go
Comment (haoming29) [inline config/config.go:891]: Since we use `viper.MergeConfig` for handling the merging, let's document this at least in the function header (Or docs) and pointing to viper docs for the merging mechanism under the hood.

**Review by haoming29 — APPROVED** [review]
LGTM overall, conditionally approved if the comments are addressed.

**Pull request merged**
